DSharpPlus/Entities/AuditLogs/AuditLogParser.cs

```diff
@@ -35,77 +13,66 @@
 
 namespace DSharpPlus.Entities.AuditLogs;
 
-internal static class AuditLogParser
+internal class AuditLogParser
 {
+    private BaseDiscordClient _client;
+    private DiscordGuild? _guild;
+    private ulong _guildId;
+    
+    internal AuditLogParser(BaseDiscordClient client, ulong guildId, DiscordGuild? guild = null)
+    {
+        _client = client;
+        _guildId = guildId;
+        _guild = guild;
+    }
+    
     /// <summary>
     /// Parses a AuditLog to a list of AuditLogEntries
     /// </summary>
-    /// <param name="guild"> <see cref="DiscordGuild"/> which is the parent of the AuditLog</param>
     /// <param name="auditLog"> <see cref="AuditLog"/> whose entries should be parsed</param>
     /// <returns>A list of <see cref="DiscordAuditLogEntry"/>. All entries which cant be parsed are dropped</returns>
-    internal static async IAsyncEnumerable<DiscordAuditLogEntry> ParseAuditLogToEntriesAsync
+    internal async IAsyncEnumerable<DiscordAuditLogEntry> ParseAuditLogToEntriesAsync
     (
-        DiscordGuild guild,
         AuditLog auditLog
     )
     {
-        BaseDiscordClient client = guild.Discord;
-
-        //Get all User
-        IEnumerable<DiscordUser> users = auditLog.Users;
-
-        //Update cache if user is not known
-        foreach (DiscordUser discordUser in users)
-        {
-            discordUser.Discord = client;
-            if (client.UserCache.ContainsKey(discordUser.Id))
-            {
-                continue;
-            }
-
-            client.UpdateUserCache(discordUser);
-        }
-
-        //get unique webhooks, scheduledEvents, threads
+        _guild ??= await _client.Cache.TryGetGuildAsync(this._guildId);
+        
+        IEnumerable<DiscordUser> users = auditLog.Users.ToList();
```

Comment:
why the ToList call?

Reply:
This comes directly from a json response an i want to be safe from multiple enumerations.

Reply:
Newtonsoft.Json parses to a list and returns that